### ISSUE TITLE
Game crashes while loading save file during firefighting attempt

### ISSUE THREAD

**Issue by zer0Entropy** [CleverRaven/Cataclysm-DDA#47614]
# Describe the bug
Save file (generated by auto-save) crashes the game with SEGFAULT upon loading.

# Steps To Reproduce
Okay.  I will attempt to keep this short, but there was a complicated chain of events that lead to my discovery of this bug, so I feel compelled to walk you through the whole thing:

After killing a Gasoline Zombie on level [...3? 4? shouldn't matter] of a Science Lab, my character endeavored to combat the roaring fires that erupted immediately after the ensuing explosion.

I grabbed a small fire extinguisher I had stashed nearby (for I had spotted this sucker earlier, and had attempted to prepare).  I emptied it into the nearest tile described as "raging fire", and lathered, rinsed, and repeated until the fire extinguisher had no more charges left.

But the fire kept burning.  I panicked, worried that the blaze might spread throughout the lab and start destroying some of the precious loot I had yet to claim.  The only container I had on hand was a small plastic bottle, but the nearest water puddle was only two steps away from the room that was on fire.

So, naturally, I proceeded to keep refilling that little bottle (using [e]xamine on the nearest water tile), walking over to the nearest tile on fire, and [u]nloading it on the ground there.  Lather, rinse, repeat.  The whole time, of course, I had my Thermal Dissipation CBM active so I wouldn't burn to death.

You may have already guessed where this is going -- my bionic energy stores are depleted, now, and the fire is not really any smaller.  So I decided to Alt-F4 out and reload from my last save.  That's when I discovered that, not only had the game autosaved in the midst of my firefighting, but that this save file was bugged in such a way that it crashes every time the game tries to load it.

I am unsure exactly what I did in the above scenario that resulted in an unloadable save file, but my crash.log (see attached) seems to implicate the function "unload_activity_actor" (which I understand was added in recent pull request) in the line:
(libbacktrace: ZN21unload_activity_actor6unloadER9CharacterR13item_location+0x31@0x433421)

That's as far as my debugging capabilities, such as they are, have allowed me to narrow it down to the source of the problem.

# Expected behavior
The game should reliably load the game from any save files it generated without crashing.

# Screenshots

# Versions and configuration
VERSION: 0.E-9739-g44d04c5
Windows 10, Tiles Version
Language: English
Mods loaded: Dark Days Ahead, Disable NPC Needs, Bionic Professions, Graphical Overmap, SpeedyDex

# Additional context
[crash.log](https://github.com/CleverRaven/Cataclysm-DDA/files/6012992/crash.log)
[debug.log](https://github.com/CleverRaven/Cataclysm-DDA/files/6012993/debug.log)
[bugged_save.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/6013040/bugged_save.zip)

**Comment by wapcaplet** [comment]
Can you reproduce the issue without using Alt+F4? Quitting the game like that can lead to unexpected results.

**Comment by zer0Entropy** [comment]
> Can you reproduce the issue without using Alt+F4? Quitting the game like that can lead to unexpected results.
I'll play around with it, but I doubt I can find a way to do so reliably.  My uneducated guess, as a hobbyist coder who's barely glanced at Cataclysm's source, is that the game auto-saved while I was in the middle of unloading water out of the bottle, and for whatever reason it can't recover the game-state while my character is mid-unloading.  If that is the case, the only circumstances where this bug would occur are cases of abrupt/abnormal program shutdown (ALT-F4 or computer crashed or power failure or whatever).

I understand that ALT-F4'ing out of a situation you don't like is not the intended playstyle, nor something the dev team is interested in going out of their way to enable.  It makes perfect sense, to me, if this bug and others like it aren't counted as "release-blockers" for that reason.

I only came here to submit a bug report to help make the game more robust.  Regardless of whether this bug currently affects "legitimate gameplay", it is there.  Perhaps another related bug will crop up, and the log files I have provided will come in handy there.  (This seems particularly likely if the crash I experienced resulted from code changed in a recent pull request.)

In any case, I'll fool with it on my own, and report back if I happen to learn anything interesting.
